### PR TITLE
Clarify explanation of Vec<T> 'fn resize'

### PR DESCRIPTION
**Pull request by JOE1994**
1. Clarified on what should implement `Clone` trait.
2. Minor grammar fix:
to be able clone => to be able **to** clone

### CODE REVIEW

File: src/liballoc/vec.rs [hunk 1]
```diff
@@ -1472,8 +1472,9 @@ impl<T: Clone> Vec<T> {
     /// difference, with each additional slot filled with `value`.
     /// If `new_len` is less than `len`, the `Vec` is simply truncated.
     ///
-    /// This method requires [`Clone`] to be able clone the passed value. If
-    /// you need more flexibility (or want to rely on [`Default`] instead of
+    /// This method requires `value` to implement [`Clone`],
```
Comment (jonas-schievink) [inline src/liballoc/vec.rs]:
```suggestion
    /// This method requires `T` to implement [`Clone`], 
```

**Review by jonas-schievink — CHANGES_REQUESTED** [review]

**Comment by Dylan-DPC** [comment]
sorry, it is fine :D 

@bors r+ rollup

**Pull request merged**